Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Overrides/PrefabOverrideHandler.cpp

```diff
@@ -107,5 +77,200 @@
+                    }
+
+                    // Apply individual overrides to target prefab template.
+                    PrefabUndoComponentPropertyEdit* state = aznew PrefabUndoComponentPropertyEdit("Apply Override");
+                    state->SetParent(undoBatch.GetUndoBatch());
+                    state->Capture(targetInstance->get(), AZ::Dom::Path(overridePathStr).ToString(), metaData.m_patch.FindMember("value")->value);
+                    state->Redo();
+
+                    return true;
+                },
+                AZ::Dom::PrefixTreeTraversalFlags::None
+            );
+
+            // Create node for overrides reversal so that they get restored on undo.
+            PrefabUndoRevertOverrides* state = new Prefab::PrefabUndoRevertOverrides("Capture Override SubTree");
+            state->Capture(path, AZStd::move(subTree), linkId);
+            state->SetParent(undoBatch.GetUndoBatch());
+
+            // Queue a refresh of the property display.
+            AzToolsFramework::ToolsApplicationEvents::Bus::Broadcast(
+                &AzToolsFramework::ToolsApplicationEvents::Bus::Events::InvalidatePropertyDisplay,
+                AzToolsFramework::Refresh_AttributesAndValues);
+
+            return true;
+        }
+
+        bool PrefabOverrideHandler::PushOverrideToLink(
+            const AZ::Dom::Path& path,
+            AZStd::string_view relativePath,
+            LinkId sourceLinkId,
+            LinkId targetLinkId
+        ) const
+        {
+            // Retrieve link.
+            LinkReference sourceLink = m_prefabSystemComponentInterface->FindLink(sourceLinkId);
+            LinkReference targetLink = m_prefabSystemComponentInterface->FindLink(targetLinkId);
+
+            // Remove relativePath from pathStr so that the patches from sourceLink can be applied to targetLink.
+            AZStd::string pathStr = path.ToString();
+            if (pathStr.starts_with(relativePath))
+            {
+                pathStr = pathStr.substr(relativePath.length());
+            }
+
+            // Start an undo batch.
+            ScopedUndoBatch undoBatch("Apply Overrides to Prefab");
+
+            // Remove Overrides in sourceLink.
+            PrefabOverridePrefixTree subTree = sourceLink->get().RemoveOverrides(path);
+            if (subTree.IsEmpty())
+            {
+                AZ_Warning("Prefab", false, "PrefabOverrideHandler::PushOverrides is called on a path that has no overrides.");
+                return false;
+            }
+
+            // Create a new overrides PrefabOverridePrefixTree that will be altered and passed to the target link.
+            PrefabOverridePrefixTree newOverrides;
+
+            subTree.VisitPath(
+                AZ::Dom::Path(""),
+                [&newOverrides, relativePath](
+                    [[maybe_unused]] const AZ::Dom::Path& path, [[maybe_unused]] Link::PrefabOverrideMetadata& metaData) -> bool
+                {
+                    // Copy the patch.
```

Comment:
I think it would be better if we removed these lines of comments and instead have a description about why we should call VisitPath for assembling the new override path.